### PR TITLE
Use beginSubcases() more aggressively

### PR DESCRIPTION
**Pull request by ben-clayton**
And reduce a few arrays in excessive multi-dimensional parameterizations.

This PR targets the tests that have a lot of parameterizations without using sub-cases. A treemap highlights the difference (colors represent number of non-subcased parameterizations):

| Before | After |
|----------|----------|
| before | after |

Reduces the total number of WebGPU test cases from `112,554` to `56,228`, which is almost exactly a 50% reduction.

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/execution/shader_io/workgroup_size.spec.ts [hunk 1]
```diff
@@ -47,12 +47,14 @@ function checkResults(
...
 }
 
-const kSizes = [1, 2, 3, 4, 8, 11, 16, 51, 64, 128, 256];
-
 g.test('workgroup_size')
   .desc(`Test workgroup size is set correctly`)
   .params(u =>
-    u.combine('wgx', kSizes).combine('wgy', kSizes).combine('wgz', kSizes).beginSubcases()
+    u
+      .combine('wgx', [1, 3, 4, 8, 11, 16, 51, 64, 128, 256] as const)
```
Comment (zoddicus) [inline src/webgpu/shader/execution/shader_io/workgroup_size.spec.ts:54]: Should the .combines here be after the .beginSubcases?

**Review by zoddicus — APPROVED** [review]

File: src/webgpu/shader/execution/shader_io/workgroup_size.spec.ts (hunk 1 above)
Comment (ben-clayton) [reply src/webgpu/shader/execution/shader_io/workgroup_size.spec.ts:54]: I decided against that, given that the test is specifically testing the various workgroup sizes. I just reduced the O(n<sup>3</sup>) complexity to something more sensible.

Comment (zoddicus) [reply src/webgpu/shader/execution/shader_io/workgroup_size.spec.ts:54]: sgtm

**Review by kainino0x — APPROVED** [review]
LGTM. Great tool.

**Pull request merged**